src/ReinforcementLearningZoo/src/algorithms/nfsp/nfsp_example.jl

```diff
@@ -0,0 +1,140 @@
+export NFSPAgent
+
+using Distributions: TruncatedNormal
+
+"""
+    NFSPAgent(; η, rng, rl_agent::Agent, sl_agent::Agent)
+
+Neural Fictitious Self-Play (NFSP) agent implemented in Julia. 
+See the paper https://arxiv.org/abs/1603.01121 for more details.
+
+# Keyword arguments
+
+- `rl_agent::Agent`, Reinforcement Learning(RL) agent(use `DQN` as follows), which works to search the best response from the self-play process.
+- `sl_agent::Agent`, Supervisor Learning(SL) agent(use `BehaviorCloningPolicy` as follows), which works to learn the best response from the rl_agent's policy.
+- `η`, anticipatory parameter, the probability to use `ϵ-greedy(Q)` policy when training the agent.
+- `rng=Random.GLOBAL_RNG`.
+- `update_freq::Int`: the frequency of updating the agents' `approximator`.
+- `step_counter::Int`, count the step.
+"""
+mutable struct NFSPAgent <: AbstractPolicy
+    rl_agent::Agent
+    sl_agent::Agent
+    η
+    rng
+    update_freq::Int
+    step_counter::Int
+    mode::Bool
+end
+
+function NFSPAgent(
```

Comment:
Although this constructor is somewhat useful, it has a lot of constraints. And it may lead to many breaking changes in the future. Can we just move this part into RLExperiments? Or just drop it?